narwhals/_pandas_like/utils.py

```diff
@@ -567,9 +570,25 @@ def narwhals_to_native_dtype(  # noqa: PLR0915
     if isinstance_or_issubclass(dtype, dtypes.Enum):
         msg = "Converting to Enum is not (yet) supported"
         raise NotImplementedError(msg)
-    if isinstance_or_issubclass(dtype, dtypes.List):  # pragma: no cover
-        msg = "Converting to List dtype is not supported yet"
-        return NotImplementedError(msg)
+    if isinstance_or_issubclass(dtype, dtypes.List):
+        if implementation is Implementation.PANDAS and backend_version >= (2, 2):
+            import pandas as pd  # ignore-banned-import
+            import pyarrow as pa  # ignore-banned-import
```

Comment:
separately to this PR, but when we do a banned import, we should probably do a try-except and raise a message from narwhals if it fails